src/components/Auth/RegisterCheckBox.js

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useCallback } from 'react';
+import * as S from 'components/Auth/styles';
+import styled from 'styled-components';
+
+const CheckDotLabel = styled.label`
+  display: inline-block;
+  vertical-align: middle;
+  align-items: center;
+  cursor: pointer;
+`;
+
+const CheckDot = styled.input.attrs({ type: 'checkbox' })`
+  display: none;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  white-space: nowrap;
+`;
+
+const CustomCheckDot = styled.div`
+  display: inline-block;
+  width: 13px;
+  height: 13px;
+  border-radius: 50%;
+  margin-right: 20px;
+  border: solid 1px #c4c4c4;
+  transition: all 150ms;
+  ${CheckDot}:checked + & {
+    border: solid 1px #ffd25d;
+    background-color: #ffd25d;
+  }
+`;
+
+const RegisterCheckBox = ({ checked, ...props }) => {
+  const [checkedList, setCheckedLists] = useState([]);
+  const dataLists = [
+    { id: 1, text: '개인정보 이용약관(필수)' },
+    { id: 2, text: 'koala 이용약관(필수)' },
+  ];
+
+  const onCheckedAll = useCallback(
+    (checked) => {
+      if (checked) {
+        const checkedListArray = [];
+        dataLists.forEach((list) => checkedListArray.push(list.id));
+        setCheckedLists(checkedListArray);
+      } else {
+        setCheckedLists([]);
+      }
+    },
+    [dataLists]
```

Comment:
의존성으로 dataLists 를 사용하셨는데, `dataLists` 는 상태가 아닌 일반 변수입니다.
의존성을 둬야 할 의미가 없어보입니다.
```suggestion
    []
```

Reply:
3f2d393